### DISCUSSION TITLE
Get hold of the field name in Generator.

### DISCUSSION THREAD

**Discussion by NicklasWallgren** [instancio/instancio#810] [Q&A] [answered]
Is it possible to retrieve the field name when generating values?

I wan't all my string to use the following "template";  
`<name of field> <random number>`

```java
record User(String firstName, String lastName){}

var user = Instancio.of(User.class)
    // How do I get hold of the field name in the generator?
    .supply(allStrings(), (gen) -> "%s - %s".format("<name of field>", gen.digits(5)))
    .create();
```

Expected result
```
User[firstName=firstName - 79864, lastName=lastName - 79349]
```

**Comment by armandino** [comment]
@NicklasWallgren - it's not possible to retrieve the field name as you described, but there is a setting `Keys.STRING_FIELD_PREFIX_ENABLED` which does something similar:

```java
var user = Instancio.of(User.class)
    .withSettings(Settings.create().set(Keys.STRING_FIELD_PREFIX_ENABLED, true))
    .create();
```

**Reply by NicklasWallgren** [reply]
Great, thank you!

**Reply by NicklasWallgren** [reply]
@armandino It didn't work unfortunately.

```
var user = Instancio.of(User.class)
            .withSettings(Settings.create().set(Keys.STRING_FIELD_PREFIX_ENABLED, true))
            .supply(allStrings(), (gen) ->  gen.digits(5))
            .create(); 
```

Resulted in;
`User[firstName=83624, lastName=61525]`

**Reply by armandino** [reply, marked as answer]
@NicklasWallgren - sorry, forgot to mention, if you set the string to a custom value, then the field name prefix will not be added. In the original example using `allStrings()` was setting all strings to the given value. For this reason, there was no prefix. I edited my answer above to correct the example.

Also, in your last example, when you use `supply()`, you get an instance of `org.instancio.Random`, not `Generators`, i.e.

```java
supply(selector, (Random random) ->  ...)

generate(selector, (Generators gen) -> ...)
```

Hope this helps.
